### PR TITLE
Clean up client

### PR DESCRIPTION
**Pull request by mason-fish**
Some housekeeping with the Makefile, and fixing the corresponding `fmt`, `vet`, and `golint` warnings included.

Also includes both a detailed and lighter version of the `ListSubscriptions()` method for more meaningful consumption by clients of this lib.

Also includes some code style changes with regards to named/naked returns.

### CODE REVIEW

File: vendor/github.com/docker/docker/LICENSE [hunk 1]
```diff
@@ -0,0 +1,191 @@
+
```
Comment (tescherm) [inline vendor/github.com/docker/docker/LICENSE]: does `docker/docker` need to be vendored in? If possible we should remove this to eliminate cyclic dependencies.

File: client.go [hunk 2]
```diff
@@ -16,23 +16,26 @@ import (
...
+// licensing, accounts, and billing services
 type Client interface {
 	LoginViaAuth(ctx context.Context, username, password string) (authToken string, err error)
 	GetHubUserOrgs(ctx context.Context, authToken string) (orgs []model.Org, err error)
 	GetHubUserByName(ctx context.Context, username string) (user *model.User, err error)
 	VerifyLicense(ctx context.Context, license model.IssuedLicense) (res *model.CheckResponse, err error)
 	GenerateNewTrialSubscription(ctx context.Context, authToken, dockerID, email string) (subscriptionID string, err error)
-	ListSubscriptions(ctx context.Context, authToken, dockerID string) (response []*model.SubscriptionDetail, err error)
+	ListSubscriptions(ctx context.Context, authToken, dockerID string) (response []*model.Subscription, err error)
+	ListDetailedSubscriptions(ctx context.Context, authToken, dockerID string) (response []*model.SubscriptionDetail, err error)
```
Comment (dhiltgen) [inline client.go]: 'nit - I would have named it `ListSubscriptionDetails` to be symmetric with the structure name.

File: client.go [hunk 3]
```diff
@@ -16,23 +16,26 @@ import (
 )
 
 const (
-	TRIAL_PRODUCT_ID   = "docker-ee-trial"
-	TRIAL_RATE_PLAN_ID = "free-trial"
+	trialProductID  = "docker-ee-trial"
+	trialRatePlanID = "free-trial"
```
Comment (dhiltgen) [inline client.go]: Longer term I think we'll want to be able to generate different types of trials, although I'm not sure yet what the best model is to expose that choice to the library client.   For now locking it down to this one type of trial is probably fine.

File: client.go [hunk 4]
```diff
@@ -129,6 +132,30 @@ func (c *client) ListSubscriptions(ctx context.Context, authToken, dockerID stri
 		})
 	}
 
+	// filter out non docker licenses
+	dockerSubs := []*model.Subscription{}
+	for _, sub := range subs {
+		if !strings.HasPrefix(sub.ProductID, "docker-ee") {
+			continue
+		}
```
Comment (dhiltgen) [inline client.go]: This seems like an area we'll want to revisit as we generalize this, but for now this seems OK.

File: vendor/github.com/docker/docker/LICENSE (hunk 1 above)
Comment (dhiltgen) [reply vendor/github.com/docker/docker/LICENSE]: `storage.go` requires it.

If this turns out to be problematic, we might want to make a sub-package for storage so clients that don't need storage can omit it (although all the clients I can think of at the moment will want it, so maybe it's moot.)

**Pull request merged**